### PR TITLE
feat: migrate dust apps v1 endpoints to vaults

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/tasks/issues/1256

Moves the v1 endpoints for dust apps under `/vault`. Legacy endpoints are kept (they use the same handler as the new one).
Only legacy endpoints are documented for now. 

## Risk

Changing the whole permissions-checking logjc on dust apps public endpoint. 

## Deploy Plan

N/A

### CODE REVIEW

File: front/lib/resources/group_resource.ts [hunk 1]
```diff
@@ -125,7 +125,7 @@ export class GroupResource extends BaseResource<GroupModel> {
       where: {
         workspaceId: key.workspaceId,
         [Op.or]: [
-          { kind: key.isSystem ? "system" : "global" },
+          { kind: key.isSystem ? ["system", "global"] : "global" },
```
Comment (fontanierh) [inline front/lib/resources/group_resource.ts:128]: Is this correct @spolu @tdraier  ? System key should have access to both system and global groups.

**Review by tdraier — COMMENTED** [review]
why is it disabled for all other vaults than global ?

**Comment by fontanierh** [comment]
> why is it disabled for all other vaults than global ?

Because API key can only manage the global vault

**Review by tdraier — APPROVED** [review]
LGTM 👍

**Comment by fontanierh** [comment]
Re-tested following last minute changes. good to go

File: front/lib/resources/group_resource.ts (hunk 1 above)
Comment (spolu) [reply front/lib/resources/group_resource.ts:128]: That looks not crazy.

cc @flvndvd given the TODO(GROUPS_INFRA)

**Pull request merged**